### PR TITLE
Modified the navbar to look nice.

### PR DESCRIPTION
**Pull request by SoumyadiptoPal**
For small devices a right drawer is made for navigation. Also no overflow in navbar.

### CODE REVIEW

**Comment by SoumyadiptoPal** [comment]
image

File: src/components/navbar/Navbar.jsx [hunk 1]
```diff
@@ -25,113 +25,64 @@ const navbarLinks = [
...
-          zIndex: 100,
-        }}
+
+  return (
+    <nav
+      className="flex flex-row justify-between p-3"
+      style={{
+        backgroundColor: "#121930",
+        padding: "10px 60px",
+        alignItems: "center",
```
Comment (AdityaMayukhSom) [inline src/components/navbar/Navbar.jsx:38]: Please remove the styles which are present in tailwindcss, and add the corresponding tailwindcss class instead.

**Review by AdityaMayukhSom — COMMENTED** [review]
### 1. Slight modification in UI

image

The spacing between the heading and hamburger is not consistent. They should be further seperated.

Use [this](https://www.flaticon.com/free-icon/align-text_10309093?term=menu-burger&page=2&position=38&origin=tag&related_id=10309093) hamburger icon if possible.

### 2. Base page is scrollable when sidebar is opened

image

Use a backdrop (opaque dark layer) to cover the base page and disable scrolling when sidebar is opened.

### 3. Change color of sidebar

image

Please change color of sidebar, to probably the base color of github dark theme website, or `rgb(18, 25, 48)`, whichever looks better. Also the sidebar should take move screen width, try 80%, 70% and 60%, then use the one which looks better.

image

You can also use some icons and the color change of the selected section like this.
To change the highlight when simply scrolling without the use of sidebar, you can use [IntersectionObserver API](https://www.freecodecamp.org/news/reveal-on-scroll-in-react-using-the-intersection-observer-api/#:~:text=Intersection%20Observer%20is%20a%20web,or%20exits%20%2F%20enters%20the%20viewport.) if required.

### 4. Try using CSS to handle the open and closing of navbar

[Reference](https://www.youtube.com/watch?v=XM7sEpl0f7c)

Try to not use window object directly as thet might cause error in production if the javascript is executed before the DOM is completely loaded. Try to use CSS media query and CSS to handle the transition, also that will help you to have a transition effect for the sidebar. 

The event listener on window will be mostly idle in all production cases because window resize events are only triggered when the user resizes the browser, that too in desktop environment, otherwise there is no window resize in mobile. 

Also if you want to trigger something when the window.innerWidth changes, you should do that by using an useEffect hook with window.innerWidth as a dependency parameter, via passing that into the dependency array. So my suggestion is to avoid that and use simple CSS for the implementation.


**Note** : I'll add Jest into this repository, please pull the repository once, then write some tests for it so if in future you modify this section, the base functionality can be checked before deployment.

**Review by AdityaMayukhSom — CHANGES_REQUESTED** [review]
Please do the changes requested.

**Pull request merged**